### PR TITLE
Motion Matching: .CSV serializers for skeletal poses, query vectors and best matching frames

### PR DESCRIPTION
**Pull request by amzn-jillich**
CSV exporters for skeletal poses, motion matching query vectors and best matching frames

### CODE REVIEW

**Review by hultonha — APPROVED** [review]
Overall looks like a solid change, I just have a few questions/comments but nothing that's blocking, this still looks like a very useful change to get in for Motion Matching

File: Gems/MotionMatching/Code/Source/CsvSerializers.cpp [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+        {
+            return false;
+        }
+
+        m_tempBuffer = "Best Matching Frames";
+        WriteLine(m_tempBuffer);
+        return true;
+    }
+
+    void BestMatchingFrameWriterCsv::Write(size_t frameNr)
```
[the remark this answers is not in the document]
Comment (amzn-jillich) [reply Gems/MotionMatching/Code/Source/CsvSerializers.cpp:213]: Good call, changed

File: Gems/MotionMatching/Code/Source/CsvSerializers.h [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+{
+    class ActorInstance;
+};
+
+namespace EMotionFX::MotionMatching
+{
+    class FeatureSchema;
+    class QueryVector;
+
+    class EMFX_API CsvWriterBase
```
[the remark this answers is not in the document]
Comment (amzn-jillich) [reply Gems/MotionMatching/Code/Source/CsvSerializers.h:24]: Added

File: Gems/MotionMatching/Code/Source/CsvSerializers.h [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    //! get the Z axis, create a rotation matrix from that and then convert it back to a quaternion.
+    class EMFX_API PoseWriterCsv
+        : public CsvWriterBase
+    {
+    public:
+        virtual ~PoseWriterCsv() = default;
+
+        bool Begin(const char* filename, ActorInstance* actorInstance);
+        void WritePose(Pose& pose);
+        void End();
```
[the remark this answers is not in the document]
Comment (amzn-jillich) [reply Gems/MotionMatching/Code/Source/CsvSerializers.h:58]: Yeah, good call. Added that

File: Gems/MotionMatching/Code/Source/CsvSerializers.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (hultonha) [reply Gems/MotionMatching/Code/Source/CsvSerializers.cpp]: Nice one thanks! :+1:

**Pull request merged**